Firestore/core/src/remote/serializer.cc

```diff
@@ -1115,21 +1115,21 @@ google_firestore_v1_Cursor Serializer::EncodeBound(const Bound& bound) const {
   result.before = bound.before();
   SetRepeatedField(
       &result.values, &result.values_count,
-      absl::Span<google_firestore_v1_Value>(bound.position().values,
-                                            bound.position().values_count),
+      absl::Span<google_firestore_v1_Value>(bound.position()->values,
+                                            bound.position()->values_count),
       [](const google_firestore_v1_Value& value) { return DeepClone(value); });
   return result;
 }
 
 Bound Serializer::DecodeBound(google_firestore_v1_Cursor& cursor) const {
-  google_firestore_v1_ArrayValue index_components;
-  SetRepeatedField(&index_components.values, &index_components.values_count,
+  SharedMessage<google_firestore_v1_ArrayValue> index_components{{}};
```

Comment:
This initialization looks a little weird (though it does the right thing, AFAICT). Consider adding a named "constructor" for this case, something like:
```cpp
auto index_components = SharedMessage<google_firestore_v1_ArrayValue>::CreateDefault();
```

The name can definitely be changed (maybe it should be `Empty` or something).

Applies to the other files as well.

Reply:
I used the new MakeSharedMessage function. I like the explicit nature a bit better as a `SharedMessage<google_firestore_v1_ArrayValue>::Empty()` factory might suggest that it returns the same singleton instance.